src/node/rpc/frontend.h

```diff
@@ -361,7 +363,8 @@ namespace ccf
           // compaction. Reset and retry
           LOG_DEBUG_FMT(
             "Transaction execution conflicted with compaction: {}", e.what());
-          tx.reset();
```

Comment:
If I remember correctly, the only reason we had `.reset()` was because `Tx` had no copy-constructor. But the desired behaviour was "set this back to a freshly constructed state", and we had to that by keeping the implementation of `.reset()` in-sync has new members were added? Instead, I've added a move-assign operator to the `Tx`, so now we can do this reset more consistently as a "build me a new one" (but we still have no copy-constructor, so the compiler encourages us to pass-by-ref and not accidentally copy the `Tx`).